src/Core/Traits/PermissionAccess.php

```diff
@@ -43,14 +44,14 @@ protected function hasRolesEnabled()
 	 * Check if the user has permission
 	 * @return boolean
 	 */
-	protected function hasPermission($user)
+	protected function hasPermission(User $user, $permission)
 	{
 		// Don't check for permissions if we don't have the
 		// roles feature enabled
 		if (!$this->hasRolesEnabled()) {
```

Comment:
Yeah that would actually be a nice approach it would be neat to separate roles into its own acl space